packages/react/src/lazy.spec.tsx

```diff
@@ -285,4 +282,112 @@ describe('lazy', () => {
       expect(onError).toHaveBeenCalledWith({ error: expect.any(Error), load: expect.any(Function) })
     })
   })
+
+  describe('reloadOnError option', () => {
```

Comment:
Can we define specs for below case?

```tsx
import { lazy } from '@suspensive/react'

const customLazy = lazy.create(reloadOnError({ ... }))
const Comp = customLazy(() => import('...'), { onSuccess, onError }) // I expect this don't destroy reloadOnError
```

```tsx
import { lazy } from '@suspensive/react'

const Comp = lazy(() => import('...'), reloadOnError({ onSuccess, onError })) // I expect these onSuccess, onError don't destroy reloadOnError
```

```tsx
import { lazy } from '@suspensive/react'

const customLazy = lazy.create(reloadOnError({ onSuccess, onError })) // I expect these onSuccess, onError don't destroy reloadOnError
const Comp = customLazy(() => import('...')) 
```

Reply:
I’ve added the `preventMerging` option in commit 9b54bb0a06b7c24c4303fd9c823b9ac8efd94fa4. \
My only concern is that this might introduce too many options for users.

Overriding `onSuccess` feels unnatural, but without the `preventMerging` option, users can’t disable the default behavior without creating a new lazy instance.

Please feel free to giving me feedbacks, @manudeli @gwansikk @kangju2000

Reply:
Thank you for the long discussion! I think we finally found the best direction.